src/contexts/states/logExportStore.ts

```diff
@@ -0,0 +1,63 @@
+import {create} from "zustand";
+
+import LogExportManager, {EXPORT_LOGS_PROGRESS_VALUE_MIN} from "../../services/LogExportManager";
+import {Nullable} from "../../typings/common";
+import {LOG_LEVEL} from "../../typings/logs";
+import {DO_NOT_TIMEOUT_VALUE} from "../../typings/notifications";
+import {EXPORT_LOGS_CHUNK_SIZE} from "../../utils/config";
+import useContextStore from "./contextStore";
+import useLogFileManagerProxyStore from "./logFileManagerProxyStore";
+import useLogFileStore from "./logFileStore";
+
+
+interface LogExportValues {
+    exportProgress: Nullable<number>;
+    logExportManager: Nullable<LogExportManager>;
+}
+
+interface LogExportActions {
+    setExportProgress: (newProgress: Nullable<number>) => void;
+
+    exportLogs: () => void;
+}
+
+type LogExportState = LogExportValues & LogExportActions;
+
+const LOG_EXPORT_STORE_DEFAULT: LogExportValues = {
+    exportProgress: null,
+    logExportManager: null,
+};
+
+const useLogExportStore = create<LogExportState>((set) => ({
+    ...LOG_EXPORT_STORE_DEFAULT,
+    setExportProgress: (newProgress) => {
+        set({exportProgress: newProgress});
+    },
+    exportLogs: () => {
+        const {postPopUp} = useContextStore.getState();
+        const {logFileManagerProxy} = useLogFileManagerProxyStore.getState();
+        const {fileName, numEvents} = useLogFileStore.getState();
+
+        set({exportProgress: EXPORT_LOGS_PROGRESS_VALUE_MIN});
+        const logExportManager = new LogExportManager(
+            Math.ceil(numEvents / EXPORT_LOGS_CHUNK_SIZE),
+            fileName
+        );
+
+        set({logExportManager});
+        (async () => {
+            await logFileManagerProxy.exportLogs();
+        })().catch((e: unknown) => {
+            console.error(e);
+            postPopUp({
+                level: LOG_LEVEL.ERROR,
+                message: String(e),
+                timeoutMillis: DO_NOT_TIMEOUT_VALUE,
+                title: "Action failed",
+            });
+        });
+    },
+}));
```

Comment:
as discussed offline, let move the state imports right before their usages. e.g.,

```suggestion
    exportLogs: () => {
        set({exportProgress: EXPORT_LOGS_PROGRESS_VALUE_MIN});

        const {fileName, numEvents} = useLogFileStore.getState();
        const logExportManager = new LogExportManager(
            Math.ceil(numEvents / EXPORT_LOGS_CHUNK_SIZE),
            fileName
        );

        set({logExportManager});
        (async () => {
            const {logFileManagerProxy} = useLogFileManagerProxyStore.getState();
            await logFileManagerProxy.exportLogs();
        })().catch((e: unknown) => {
            console.error(e);

            const {postPopUp} = useContextStore.getState();
            postPopUp({
                level: LOG_LEVEL.ERROR,
                message: String(e),
                timeoutMillis: DO_NOT_TIMEOUT_VALUE,
                title: "Action failed",
            });
        });
    },
}));
```

Reply:
discussed offline: since the logic is already correct, where to put the state imports is simply an ordering issue and is stylish. Importing states right before we need to reference them will certainly help reducing issues from circular dependencies of Zustand actions, but we don't have such dependency issues in our current code as well.

we will reorganize the statements in anther PR. @coderabbitai create an issue to track this